src/txn/txn.c

```diff
@@ -655,11 +655,18 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 /*
  * __txn_commit_timestamps_assert --
  *	Validate that timestamps provided to commit are legal.
+ *
+ *	For prepared transactions, this function will open cursors to mutate the
+ *	update list with pointers to uncommitted data. After they are finished
+ *	with the update list, the caller is responsible for closing every
+ *	non-null cursor pointer in the cursors array and freeing the underlying
+ *	buffer.
  */
 static inline int
-__txn_commit_timestamps_assert(WT_SESSION_IMPL *session)
+__txn_commit_timestamps_assert(
+    WT_SESSION_IMPL *session, WT_CURSOR ***cursor_buf)
```

Comment:
http://wiki.c2.com/?ThreeStarProgrammer

Reply:
Ha ha! We don't have a lot of them, but we do have a few. `os_posix/os_dir.c, btree/bt_split.c` for a couple examples. The comment is very helpful.